docs/acp/user/app-kernel.md

```diff
@@ -0,0 +1,126 @@
+
+- Configure an App (receive a request from the App Store Catalog to configure this App for a Tenant)
+- Disconnect an App (an App gets disconnected from a Tenant through the App Store Catalog)
+
+This guide provides a step-by-step description of how to use the AppKernel together with Spryker's Mini Framework to develop an App.
+
+## AppKernel
+Basically, every App needs to be configured for each Tenant individually. To prevent this logic from being implemented with each PBC the AppKernel provides this functionality out of the box.
+
+This includes:
+
+- The route provider plugin \Spryker\Glue\App\Plugin\RouteProvider\AppRouteProviderPlugin which provides Glue endpoints
+  - `/private/configure` - Used to receive the configuration request.
+  - `/private/disconnect` - Used to disconnect the App from a Tenant (configuration gets deleted).
+- The controller \Spryker\Glue\App\Controller\AppConfigController::postConfigureAction() which will receive the configuration request. 
+- The controller \Spryker\Glue\App\Controller\AppDisconnectController::postDisconnectAction() which will receive the disconnection request. 
+ 
+Additionally, you can extend the AppKernel with your own business logic by using plugins in the provided extension points as outlined in the integration guide.
+
+## Install the required modules using Composer
+
+```bash
+composer require spryker/app-kernel
+```
+
+
+## Configure routes
+
+Add the `AppRouteProviderPlugin` to `\Pyz\Glue\GlueBackendApiApplication\GlueBackendApiApplicationDependencyProvider::getRouteProviderPlugins()`.
+
+## Extending Glue
+
+In many cases, it is required to add your own validation on top of the default validation that is provided by the AppKernel. We offer extension points for both actions Configure and Disconnect. You can use the same plugin in both places when needed.
+
+### Configure
+
+Add a class that implements `\Spryker\Glue\GlueApplicationExtension\Dependency\Plugin\RequestValidatorPluginInterface` and add it to `\Pyz\Glue\AppKernel\AppKernelDependencyProvider::getRequestConfigureValidatorPlugins()` on the project level when needed.
+
+This plugin will be executed inside the `\Spryker\Glue\AppKernel\Controller\AppConfigController::postConfigureAction()` before any other action is happening.
+
+### Disconnect
+
+Add a class that implements `\Spryker\Glue\GlueApplicationExtension\Dependency\Plugin\RequestValidatorPluginInterface` and add it to `\Pyz\Glue\AppKernel\AppKernelDependencyProvider::getRequestDisconnectValidatorPlugins()` on the project level when needed.
+
+This plugin will be executed inside the `\Spryker\Glue\AppKernel\Controller\AppConfigController::postDisconnectAction()` before any other action is happening.
+
+## Extending Zed
+
+### Configure
+
+In some cases, you need to do additional things with the passed configuration e.g. sending a request to a third-party application, or manipulating the data before it gets saved into the database.
+
+You can add two plugins:
```

Comment:
where should they be added?

Reply:
`\Pyz\Zed\AppKernel\AppKernelDependencyProvider::getConfigurationBeforeSavePlugin()`
`\Pyz\Zed\AppKernel\AppKernelDependencyProvider::getConfigurationAfterSavePlugin()`